webpack/src/Extends/Host/PuppetTab/SubTabs/Reports/components/ConfigStatusCard.js

```diff
@@ -0,0 +1,179 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect } from 'react';
+import {
+  Divider,
+  Flex,
+  FlexItem,
+  Button,
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+} from '@patternfly/react-core';
+import {
+  TableComposable,
+  TableText,
+  Tr,
+  Tbody,
+  Td,
+} from '@patternfly/react-table';
+import { useSelector, useDispatch } from 'react-redux';
+import { selectAPIResponse } from 'foremanReact/redux/API/APISelectors';
+import CardTemplate from 'foremanReact/components/HostDetails/Templates/CardItem/CardTemplate';
+import RelativeDateTime from 'foremanReact/components/common/dates/RelativeDateTime';
+import SkeletonLoader from 'foremanReact/components/common/SkeletonLoader';
+import DefaultLoaderEmptyState from 'foremanReact/components/HostDetails/DetailsCard/DefaultLoaderEmptyState';
+import { statusFormatter } from 'foremanReact/components/HostDetails/Tabs/ReportsTab/helpers';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import { getReportByIdAction } from './ConfigStatusCardActions';
+import { FOREMAN_PUPPET_LAST_REPORT_KEY } from './ConfigStatusCardConstants';
+
+const cardHeaderDivider = () => (
+  <Divider
+    orientation={{
+      default: 'vertical',
+    }}
+    inset={{ default: 'insetMd' }}
+  />
+);
+
+const generateCardHeader = (hostName, allReports = [], reportsCount) =>
+  reportsCount > 0 ? (
+    <>
+      {__('Last configuration status')}
+      <Flex>
+        <FlexItem>
+          <Button
+            ouiaId="foreman-puppet-last-report-button"
+            variant="link"
+            component="a"
+            isInline
+            isDisabled={allReports === []}
+            href={`/config_reports/${allReports[0].id}`}
+          >
+            <RelativeDateTime
+              date={allReports[0].reported_at}
+              defaultValue={__('Never')}
+            />
+          </Button>
+        </FlexItem>
+        <FlexItem>{statusFormatter('failed', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('failed_restarts', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('restarted', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('applied', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('skipped', allReports[0])}</FlexItem>
+        {cardHeaderDivider()}
+        <FlexItem>{statusFormatter('pending', allReports[0])}</FlexItem>
+      </Flex>
+    </>
+  ) : (
+    <> {__('No configuration status available')} </>
+  );
+
+const createPuppetMetricsTableElement = (name, value) => (
+  <>
+    <Td modifier="truncate" key={`metrics-name-${name}`}>
+      <TableText>{name}</TableText>
+    </Td>
+    <Td modifier="truncate" key={`metrics-name-${value}`}>
+      <TableText>{value}</TableText>
```

Comment:
I sometimes see 0, and sometimes no value. can we have something instead of an empty space when there is no value? 
![image](https://github.com/theforeman/foreman_puppet/assets/30431079/a127f486-009e-4a41-844b-15d52929583f)

Reply:
Any preferences about the default value?

Reply:
Does "not applicable" (`n/a`) represent correctly what it means when a metric is null?

Reply:
Personally, I don't know if the value from the Puppet report can be empty at all. But for me, (n/a) sounds reasonable.

Reply:
@ares Could you help us out here? It is about the implementation of the Puppet Metrics table https://user-images.githubusercontent.com/51960547/278665889-e07d582f-fa86-4861-ad21-0f6a39650e2c.png

Reply:
From the screenshot, we can also use `-` instead of n/a

Reply:
It is actually `--`, the same as on the Reports tab. That is what I currently have implemented.

Reply:
I mean something like ` <TableText>{value || --}</TableText>`

Reply:
What is the difference between `<TableText>{value || --}</TableText>` and setting the default value in function parameter?

Reply:
None, I missed the default in the function sorry

Reply:
Is there anything else I should fix?